README.md

```diff
@@ -6,41 +6,20 @@
 [![Checked with mypy][mypy-image]][mypy-url]
 [![Twitter][twitter-image]][twitter-url]
 
-Each directory within the `fideslib` directory represents a discrete package that exposes specific functionality intended to be shared across multiple fides tool codebases. Any amount of packages may be independently consumed by a given fides tool.
```

Comment:
I tried the piece wise, but in the end the only option that was available was `all`. The reason is `base` needs information from `oauth`, and `oauth` needs `cryptography`. If we really want to be able to install everything independently the way to do it would be to split things into individual pacage

Reply:
I think it's fine if fideslib packages have dependencies on one another. It just so happens to be the case that the packages that are currently implemented are highly incestuous, but that may not be true in the future. Maintaining a design pattern that allows piecewise imports feels like an important feature, as it might significantly reduce import volumes for projects that require only one or two potentially small packages.

We can revisit this in the future, when this library is more robust, but I would prefer to set the precedent now if possible.

Reply:
If we do this we end up with `pip install fidesctl` installing nothing, and extras are no longer really extras but required. That feels to me like we are forcing packages together that don't really belong together. To me it would make more sense to have separate packages in that case.

Reply:
Sorry, yes I meant `fideslib`. I think if we default `pip install fideslib` to the same as `pip install fideslib[all]` there will be no way to not get `all`. That said, I have never tried this so let me see what happens.

Reply:
`pip install fideslib` being equal to `pip install fideslib[all]` seems like an anti-pattern, as in theory they should be opposites (one installs zero dependencies, the other all). This is neither explicit or intuitive.

Reply:
Right, but a package that doesn't install anything unless you include extras is weird. It doesn't seem right.

Reply:
It's not a typical repository. In my view, fideslib has no standalone value; it's entire purpose is to expose shared code, and to do so efficiently.

Reply:
I think it is the wrong way to go about it, but I'll defer to the group here. If that is how everyone want it to work I will set it up that way.

Reply:
it would be surprising to me if `pip install fideslib` did nothing, I'd expect at least some core functionality to be installed there.

Reply:
This thread seems a bit undecided - do y'all need to hop on a design call and finalize here?

One thing I'll add is that, while the original intent of fideslib was to allow importing logic piecemeal (oauth, crypto, etc.) that hasn't been tested yet. Normally I don't like making breaking changes unless there is a strong reason to do so, but since this package hasn't been used in any of our projects yet, we could change that safely. In fact, since the packaging of fideslib didn't allow importing at all and a lot of the effort here was fixing that up and making it work, we know it's safe to modify 😄 

I'd recommend we do whatever is most intuitive and convenient for the consumers of this library but will leave it to y'all to make the call on what we prefer